packages/react-microlink/src/index.js

```diff
@@ -5,32 +5,39 @@ import {CardWrap, CardMedia, CardContent, CardEmptyState} from './components/Car
 import {getUrlPath, someProp, createApiUrl} from './utils'
 
 class Microlink extends Component {
+  constructor (props) {
+    super(props)
+    this.setData = this.setData.bind(this)
+  }
+
   componentWillMount () {
-    const {image, apiKey} = this.props
-    const imagesProps = [].concat(image)
+    if (this.props.data) return this.setData({data: this.props.data})
+
     const url = createApiUrl(this.props)
+    this.setState({loading: true})
 
-    this.setState({loading: true}, () =>
-      fetch(url, {headers: {'x-api-key': apiKey}})
-        .then(res => res.json())
-        .then(({status, data}) => {
-          const image = someProp(data, imagesProps)
-          const imageUrl = getUrlPath(image)
-          const {title, description, url, video} = data
-          const {color, background_color: backgroundColor} = image || {}
+    return fetch(url, {headers: {'x-api-key': this.props.apiKey}})
```

Comment:
I'd move `fetch` into the `setState` callback as it was before

Reply:
Maybe even a new `fetchData` method, adjusting this lifecycle method to:

```jsx
componentWillMount() {
  this.setState({loading: true}, this.fetchData)
}
```

Reply:
I was thinking if could be better call `this.fetchData` asap instead of wait the callback, what do you think?

Reply:
I'd avoid that - we want to be "in" `loading` whilst loading data, which could _possibly_ be missed if me mutate the state alongside it (after the `fetchData` method completes).

Alternatively `loading` could be `true` by default and therefore we wouldn't need to `setState` on mount 🤔  - but that's just off the top of my head, not taking into consideration where else it's being modified.